From: Jeff Mahoney <jeffm@suse.com>
Subject: [PATCH 05/12] ocfs2: actually free hb set on cluster removal

 This patch actually frees the hb set when the cluster dir is removed.

 fs/ocfs2/cluster/nodemanager.c |    1 +
 1 files changed, 1 insertion(+)

Signed-off-by: Jeff Mahoney <jeffm@suse.com>

diff -ruNpX ../dontdiff linux-2.6.15-staging1/fs/ocfs2/cluster/nodemanager.c linux-2.6.15-staging2/fs/ocfs2/cluster/nodemanager.c
--- linux-2.6.15-staging1/fs/ocfs2/cluster/nodemanager.c	2006-01-17 15:18:53.212949008 -0500
+++ linux-2.6.15-staging2/fs/ocfs2/cluster/nodemanager.c	2006-01-17 15:18:53.223947336 -0500
@@ -701,6 +701,7 @@ static void o2nm_cluster_group_drop_item
 	BUG_ON(o2nm_single_cluster != cluster);
 	o2nm_single_cluster = NULL;
 
+	o2hb_free_hb_set(cluster->cl_group.default_groups[1]);
 	for (i = 0; cluster->cl_group.default_groups[i]; i++) {
 		killme = &cluster->cl_group.default_groups[i]->cg_item;
 		cluster->cl_group.default_groups[i] = NULL;
